driver-core/src/main/com/mongodb/internal/connection/BaseCluster.java

```diff
@@ -326,14 +330,44 @@ private void logServerSelectionFailure(final ServerSelector serverSelector, fina
         }
     }
 
+    @Nullable
+    private ServerTuple selectServer(final ServerSelector serverSelector, final ClusterDescription clusterDescription) {
+        return selectServer(serverSelector, clusterDescription, this::getServer);
+    }
 
-    private ServerTuple selectRandomServer(final ServerSelector serverSelector, final ClusterDescription clusterDescription) {
-        List<ServerDescription> serverDescriptions = serverSelector.select(clusterDescription);
-        if (!serverDescriptions.isEmpty()) {
-            return getRandomServer(new ArrayList<ServerDescription>(serverDescriptions));
-        } else {
-            return null;
+    @Nullable
+    @VisibleForTesting(otherwise = PRIVATE)
```

Comment:
Q: Is this annotation ever not going to be otherwise=PRIVATE?

Reply:
Yes, all values of the `VisibleForTesting.AccessModifier` `enum` are possible: a program element may be made `public` for tests, while its actual access modifier is meant to be `private`/`protected`/none (a.k.a. package).